### PR TITLE
[CORE]Fix bug: read the table use partition function is months, throw exception

### LINKED ISSUE #560: [Bug]: [SPARK] Read the table use partition function is months, throw exception: type conversion failed
### What happened?

Create partitioned table use  partition function is months:
CREATE TABLE table (id int, data timestamp) USING arctic PARTITIONED BY(months(data));

then insert overwrite:
INSERT OVERWRITE TABLE local.db.table2 VALUES (1, timestamp(' 2022-1-1 09:00:00 '))

when select throw exception:
image

### Affects Versions

0.3.2

### What engines are you seeing the problem on?

Core, Spark

### How to reproduce

_No response_

### Relevant log output

_No response_

### Anything else

when use partition function days, this problem will not arise

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by hellojinsilei**
## Why are the changes needed?

fix #560 
Fix bug: read the table use partition function is months, throw exception

## Brief change log
1. add 0.4.0-init.sql,  table changes:
ALTER TABLE `table_metadata` modify COLUMN `properties` mediumtext COMMENT 'Table properties';

## How was this patch tested?
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

### CODE REVIEW

**Review by baiyangtx — APPROVED** [review]

File: core/src/main/java/com/netease/arctic/utils/ArcticDataFiles.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+import java.time.ZoneOffset;
+import java.time.temporal.ChronoUnit;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.TimeZone;
+import java.util.UUID;
+
+public class ArcticDataFiles {
+  public static final OffsetDateTime EPOCH = Instant.ofEpochSecond(0).atOffset(ZoneOffset.UTC);
+  public static final SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd-hh");
```
Comment (zhoujinsong) [inline core/src/main/java/com/netease/arctic/utils/ArcticDataFiles.java:26]: Use uppercase name for static variables.

File: ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/BaseOptimizeCommit.java [hunk 2]
```diff
@@ -107,7 +108,7 @@ public boolean commit(long baseSnapshotId) throws Exception {
                 maxTransactionIds.put(TablePropertyUtil.EMPTY_STRUCT, maxTransactionId);
               } else {
                 maxTransactionIds.putIfAbsent(
-                    DataFiles.data(spec, entry.getKey()), maxTransactionId);
+                        ArcticDataFiles.data(spec, entry.getKey()), maxTransactionId);
```
Comment (zhoujinsong) [inline ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/BaseOptimizeCommit.java:111]: Code indentation seems to wrong here.

**Review by zhoujinsong — COMMENTED** [review]
@hellojinsilei cc

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Pull request merged**